src/main/java/com/lorepo/icplayer/client/content/services/AdaptiveLearningService.java

```diff
@@ -149,12 +155,28 @@ public String getPageDifficulty(String pageID) {
 		return structure.getDifficultyForPage(pageID);
 	}
 
-	private void setPagesFromStepNonReportable(String pageID) {
-		List<String> pagesIDsFromNewPageStep = this.pageToSteps.getOtherStepPages(pageID);
+	private void saveAllReportablePages() {
+	    List<Page> allPages = this.playerController.getModel().getAllPages();
+ 	    for (Page page : allPages) {
+ 	        if (page.isReportable()){
+ 	            this.initiallyReportablePages.add(page);
+ 	        }
+ 	    }
+	}
 
-		for (String stepPageID : pagesIDsFromNewPageStep) {
-			this.setPageNonReportable(this.playerController.getModel().getPageById(stepPageID));
-		}
+	private void setCurrentPageAsReportable(String pageID) {
+	    IPage currentPage = this.playerController.getModel().getPageById(pageID);
+	    if (this.initiallyReportablePages.contains(currentPage)) {
+	        currentPage.setAsReportable();
+	    }
+	}
+
+	private void initAllPagesAsNonReportable() {
+        List<Page> allPages = this.playerController.getModel().getAllPages();
+        allPages.remove(0);
```

Comment:
Gdy użytkownik odpala lekcję na innej stronie niż 1 to wtedy ta strona przestaje być raportowalna